src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -190,6 +191,19 @@ internal static IEnumerable<IEdmFunctionImport> FilterFunctionsByParameterNames(
         }
 
         /// <summary>
+        /// Filters the operations by parameter names.
+        /// </summary>
+        /// <param name="functions">The operations.</param>
+        /// <param name="parameters">The list of non-binding parameter names to match.</param>
+        /// <param name="caseInsensitive">Whether case insensitive.</param>
+        /// <returns>The best matching operations based on parameters.</returns>
+        internal static IEnumerable<IEdmOperationImport> FindBestOverloadBasedOnParameters(this IEnumerable<IEdmOperationImport> functions, IEnumerable<string> parameters, bool caseInsensitive = false)

```

Comment:
This is used to find the best match if there are multiple candidate matches. The best match is defined to be the one for which all parameters match. If there are multiple possible matches and none are exact matches we can't unambiguously resolve and return an exception.

---
In reply to: [123296083](https://github.com/OData/odata.net/pull/828#discussion_r123296083) [](ancestors = 123296083)